app/models/spree/gateway/amazon.rb

```diff
@@ -86,11 +92,16 @@ def capture(amount, amazon_checkout, gateway_options={})
       if amount < 0
         return credit(amount.abs, nil, nil, gateway_options)
       end
-      order = Spree::Order.find_by(:number => gateway_options[:order_id].split("-")[0])
+      order_number, payment_number = gateway_options[:order_id].split("-", 2)
+      order = Spree::Order.find_by!(number: order_number)
+      payment = Spree::Payment.find_by!(number: payment_number)
+      authorization_id = order.amazon_transaction.authorization_id
+      capture_reference_id = "#{payment.number}-#{random_suffix}"
```

Comment:
So if I understand correctly, `capture_reference_id` is a unique identifier amazon needs per payment operation?